content/Improve your Rails App searches with Elasticsearch - Leandro Dominguez.md

```diff
@@ -0,0 +1,216 @@
+# Improve your Rails Apps searches with Elasticsearch
+
+## Introduction
+
+When it comes to apps with a lot of data, some SQL queries take more time than expected, especially when we want to search text through several model attributes. The most known examples are e-commerces, but this kind of complex queries can happen in apps like social media or any other one that implies making suggestions based on user information or interaction with the platform.
+
+In these cases, we want to retrieve data from our database given a query a user made.
+
+It would be reasonable to wait for a user query containing fields from different models’ attributes and in these cases queries start to get complex and inefficient. What happens when our models attributes and their relationships start to grow? How do we manage user typos in those queries? What happens when a user looks for ‘ps4’ instead of ‘playstation 4’?
```

Comment:
```suggestion
It would be reasonable to wait for a user query containing fields from different models’ attributes and in these cases queries start to get complex and inefficient. What happens when our model's attributes and their relationships start to grow? How do we manage user typos in those queries? What happens when a user looks for ‘ps4’ instead of ‘playstation 4’?
```